python/graphstorm/model/gnn.py

```diff
@@ -383,6 +383,23 @@ def set_decoder(self, decoder):
                     + 'match the input dimension of the decoder.'
         self._decoder = decoder
 
+    def set_decoder_dict(self, decoders):
```

Comment:
instead of adding a new method, can we reuse `set_decoder` and have it to support a dict of decoders?

Reply:
merged in new commit.